src/lll-implement.cpp

```diff
@@ -0,0 +1,115 @@
+#include <iostream>
+#include <stdio.h>
+#include <math.h>
+#include "matrix.h"
+using namespace std;
+
+//Calculate dot product of a[i] and b[j], used by GSO()
+int dotProduct(const Matrix<int>& a, int i,const Matrix<int>& b, int j, int ineachbasis)
+{
+    int dot= 0;
+    for(int k = 0; k < ineachbasis; k++)
+    {
+        dot+=(a[i][k]*b[j][k]);
+    }
+    return dot;
+}
+
+//Apply Gram-Schmidt Process, matrix b gives the Gram-Schmidt basis
+void GSO(Matrix<int>& a, Matrix<int>& b, int noofbasis, int ineachbasis, int B[], Matrix<float>& u)
```

Comment:
It is convention to have "return values" such as B and u at the front. Also, mark inputs which are not modified by const. Programming is about communicating with other people too, not only about "communicating" with computers.